### PR TITLE
Fix error when favoriting some types of properties

### LINKED ISSUE #99791: Getting error `Can't favorite invalid property. If said property was from a script and recently renamed, try saving it first.` when trying to favorite theme override properties
### Tested versions

- Reproducible in 4.4.dev5

### System information

Godot v4.4.dev5 - Windows 10.0.22631 - Multi-window, 2 monitors - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4080 (NVIDIA; 32.0.15.6094) - 13th Gen Intel(R) Core(TM) i9-13900K (32 threads)

### Issue description

I really like the new `Favorite Property` option in the inspector and I want to use it for theme overrides but it doesn't seem to work.

### Steps to reproduce

Click "Favorite Property" on any of these properties below. It shows this error message:

`Can't favorite invalid property. If said property was from a script and recently renamed, try saving it first.`

- HBoxContainer: theme_override_constants/separation 
- MarginContainer: theme_override_constants/margin_left

### Minimal reproduction project (MRP)

n/a

### PR DESCRIPTION
**Pull request by YeldhamDev**
- `script` and theme properties can now be (de)favorited without throwing errors.
- Deletable properties (such as metadata) no longer can be favorited.

### CODE REVIEW

File: editor/editor_inspector.cpp [hunk 1]
```diff
@@ -4337,17 +4338,37 @@ void EditorInspector::_set_property_favorited(const String &p_path, bool p_favor
...
+			class_name = validate_name;
+		}
+
+		// Deal with theme properties.
+		if (p_path.begins_with("theme_override_")) {
+			const String slice = p_path.get_slice("/", 1);
+			if (!slice.is_empty()) {
+				class_name = validate_name;
+			}
+		}
```
Comment (KoBeWi) [inline editor/editor_inspector.cpp:4367]: This doesn't seem to work, overrides still print errors.

Comment (YeldhamDev) [reply editor/editor_inspector.cpp]: Try again.

Comment (KoBeWi) [reply editor/editor_inspector.cpp]: There is no error now, but the property gets unfavorited after you re-select the node.

Comment (YeldhamDev) [reply editor/editor_inspector.cpp]: Third time is the charm.

**Review by KoBeWi — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**